test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -282,13 +282,94 @@ object Assertion extends AssertionVariants {
     }
 
   /**
-   * Makes a new assertion that requires an iterable contain the specified
-   * element. See [[Assertion.exists]] if you want to require an iterable to contain an element
+   * Makes a new assertion that requires a numeric value is positive.
+   */
+  def isPositive[A](implicit num: Numeric[A]): Assertion[A] =
+    Assertion.assertion("isPositive")()(actual => num.gt(actual, num.zero))
+
+  /**
+   * Makes a new assertion that requires a numeric value is non negative.
+   */
+  def nonNegative[A](implicit num: Numeric[A]): Assertion[A] =
+    Assertion.assertion("nonNegative")()(actual => num.gteq(actual, num.zero))
+
+  /**
+   * Makes a new assertion that requires a numeric value is negative.
+   */
+  def isNegative[A](implicit num: Numeric[A]): Assertion[A] =
+    Assertion.assertion("isNegative")()(actual => num.lt(actual, num.zero))
+
+  /**
+   * Makes a new assertion that requires a numeric value is non positive.
+   */
+  def nonPositive[A](implicit num: Numeric[A]): Assertion[A] =
+    Assertion.assertion("nonPositive")()(actual => num.lteq(actual, num.zero))
+
+  /**
+   * Makes a new assertion that requires a numeric value is zero.
+   */
+  def isZero[A](implicit num: Numeric[A]): Assertion[A] =
+    Assertion.assertion("isZero")()(_ == num.zero)
+
+  /**
+   * Makes a new assertion that requires a value to be equal to one of the specified values.
+   */
+  def equalToOneOf[A, B](value: A, other: A, remaining: A*)(implicit eql: Eql[A, B]): Assertion[B] = {
+    val allValues = value +: other +: remaining
+
+    Assertion.assertion("equalToOneOf")(param(allValues))(actual => allValues.exists(equals(actual, _)))
+  }
+
+  /**
+   * Makes a new assertion that requires an Iterable contain the specified
+   * element. See [[Assertion.exists]] if you want to require an Iterable to contain an element
    * satisfying an assertion.
    */
   def contains[A](element: A): Assertion[Iterable[A]] =
     Assertion.assertion("contains")(param(element))(_.exists(_ == element))
 
+  /**
+   * Makes a new assertion that requires an Iterable contain none of the
+   * specified elements.
+   */
+  def containsNoneOf[A](value: A, other: A, remaining: A*): Assertion[Iterable[A]] = {
```

Comment:
Maybe we could have a `hasIntersection` assertion that lets you assert that the intersection of two collections satisfies a particular assertion. Then you could implement these different variants in terms of that. I think I would also just have the parameter be a collection instead of using variable arguments.

Reply:
I like this idea. I will go ahead with this approach.